### DISCUSSION TITLE
linkerd2  upgrade  from stable-2.11.2  to  stable-2.11.2 error

### DISCUSSION THREAD

**Discussion by eddielishan** [linkerd/linkerd2#8326] [Q&A]
linkerd upgrade --set proxyInit.runAsRoot=true | kubectl apply --prune -l linkerd.io/control-plane-ns=linkerd -f -

error: error validating "STDIN": error validating data: [ValidationError(CustomResourceDefinition.spec.versions[0]): unknown field "deprecated" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinitionVersion, ValidationError(CustomResourceDefinition.spec.versions[0]): unknown field "deprecationWarning" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinitionVersion]; if you choose to ignore these errors, turn validation off with --validate=false

Is there a requirement for the k8s version?My k8s  version is v1.17.9

**Comment by olix0r** [comment]
@eddielishan I think this was an accidental regression. Unfortunately, it's no longer possible for us to run our test suite on Kubernetes v1.17; and it even appears that the [API docs for v1.17](https://kubernetes.io/docs/home/supported-doc-versions/) are no longer available on v1-17.docs.kubernetes.io. Updates to the v1.17 branch stopped sometime in 2020.

We'll [look into](https://github.com/linkerd/linkerd2/issues/8345) releasing a v2.11.3 removing the deprecation warning, but it's virtually impossible for us to validate the changes.

**Comment by eddielishan** [comment]
@olix0r  ok  thanks!!!
